### DISCUSSION TITLE
muforth meets STM8 eForth

### DISCUSSION THREAD

**Discussion by TG9541** [TG9541/stm8ef#463] [General]
@Eelkhoorn [announced](https://github.com/TG9541/stm8ef/issues/457#issuecomment-1485281595) that he made a [muforth](https://muforth.nimblemachines.com) based on the STM8 eForth kernel. 

> Here is the first attempt of using muforth to take over the interpreting and compiling functions of stm8ef. I tested it with two targets: MINDEV and W1209-FD. It is probably full of bugs, comments and suggestions for improvement are welcome. I tried it on a W1209-FD application for a simple timer, with the menu structure copied from the logging thermostat project, and it works fine.
> The flash space reduction is significant: kernel takes 2179 bytes, the application takes 1213 bytes, 4799 bytes left.

The repository is here: https://github.com/Eelkhoorn/stm8ef-mu

**Comment by Eelkhoorn** [comment]
Thank you for opening this discussion.
I didn't properly test the MINDEV case, there was a bug in forth.asm and the boardcore.inc was for the W1209 board only. Now there is a boardcore-MINDEV.inc and a boardcore-W1209.inc that have to be linked to the boardcore.inc files in the respective BOARD folders of stm8ef. The kernel takes 1733 bytes for MINDEV,  2179 for W1209-FD.
